### PR TITLE
Difference issue

### PR DESCRIPTION
**Pull request by MiyamotoAkira**
Solving issue #29. Use the first difference

### CODE REVIEW

**Comment by MiyamotoAkira** [comment]
Though `test_simple_list_with_dicts` shows the fix for the specific issue, I think the other two new tests are more interesting. Both `test_list_with_dicts` and `test_list_with_dicts_with_duplicates` show really unexpected results.

I think a better general solution would be to cache entries that have been matched (including their index) so they could be ignored on following difference findings. 

Maybe the code added (`_min_diff`) is enough for now, as the other two tests will have the same issues with  `_max_diff`, just giving different values.

**Comment by MiyamotoAkira** [comment]
Ah, I left the tests to document the current behaviour. Maybe a comment could be left on each test to indicate that the behaviour should be revisited

File: tests/test_compare.py [hunk 1]
```diff
@@ -113,6 +113,71 @@ def test_compare_deep_data(self):
...
+            {'a': "jjj"},
+        ]
+        actual = [
+            {'a': "zzz"},
+            {'a': "yyy"},
+            {'a': "xxx"},
+            {'a': "eee"},
+        ]
+        diff = Compare().check(expected, actual)
+        print(diff)
```
Comment (rugleb) [inline tests/test_compare.py]: Remove it pls.
PS: below too.

**Comment by rugleb** [comment]
> Ah, I left the tests to document the current behaviour. Maybe a comment could be left on each test to indicate that the behaviour should be revisited

Yes, add it with link on u comment please.

**Review by rugleb — CHANGES_REQUESTED** [review]

File: tests/test_compare.py (hunk 1 above)
Comment (MiyamotoAkira) [reply tests/test_compare.py]: ooooops

**Comment by MiyamotoAkira** [comment]
Both things done

**Pull request merged**